### PR TITLE
Prac 6 feedback

### PR DESCRIPTION
**Pull request by 1428-wsx-f1**
Please review my code @RemoWeiersmuller

### CODE REVIEW

File: prac_06/car.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        """Initialise a Car instance.
+
+        fuel: float, one unit of fuel drives one kilometre
+        """
+        self.fuel = fuel
+        self.name = name
+        self._odometer = 0
+
+    def __str__(self):
+        """Return car details as string"""
```
Comment (RemoWeiersmuller) [inline prac_06/car.py:17]: Doc strings usually have a '.' at the end of the line.

File: prac_06/used_cars.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+from prac_06.car import Car
+
+
+def main():
+    """Demo test code to show how to use car class."""
+    my_car = Car(180)
+    my_car.drive(30)
+    print(f"{my_car.name} has fuel: {my_car.fuel}")
+    print(my_car)
+    limo = Car(100)
```
Comment (RemoWeiersmuller) [inline prac_06/used_cars.py:16]: Where do you set the name of the car (self.name)?

File: prac_06/programming_language.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class ProgrammingLanguage:
+    """Programming language class to store details about a programming language"""
```
Comment (RemoWeiersmuller) [inline prac_06/programming_language.py:2]: Doc strings usually have '.' at the end.

File: prac_06/guitar.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        self.year = year
+        self.cost = cost
+
+    def __str__(self):
+        """Return guitar details as a string"""
+        return f"{self.name} ({self.year}) : ${self.cost:,.2f}"
+
+    def get_age(self):
+        """Return the guitars age"""
+        return CURRENT_YEAR - self.year
```
Comment (RemoWeiersmuller) [inline prac_06/guitar.py:20]: Nice version! it took me two lines to return the age. 👍

File: prac_06/guitars.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+def display_guitars(guitars):
+    """Display formatted list of guitars"""
+    for i, guitar in enumerate(guitars, 1):
+        vintage_string = ""
+        if guitar.is_vintage():
+            vintage_string = " (vintage)"
+        print(f"Guitar {i}: {guitar.name:>20} ({guitar.year}), worth ${guitar.cost:10,.2f}{vintage_string}")
+
+
+def get_valid_number(prompt):
```
Comment (RemoWeiersmuller) [inline prac_06/guitars.py:39]: Nice to have e separate function to get a valid user input, but was it really requested in the exercise?

File: prac_06/guitars.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+def main():
+    """Get guitars referencing the guitars class and display them"""
+    guitars = []
+    print("My Guitars!")
+    guitars.append(Guitar("Gibson L-5 CES", 1922, 16035.40))
+    guitars.append(Guitar("Line 6 JTV-59", 2010, 1512.9))
+    name = input("Name: ")
+    while name != "":
+        prompt = "Year: "
```
Comment (RemoWeiersmuller) [inline prac_06/guitars.py:18]: why do you define a variable for prompt? I wouldn't it be easier to just enter the string directly to the function? like: get_valid_number("Year: ") and get_valid_number("Cost: ")

File: prac_06/guitars.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        name = input("Name: ")
+
+    display_guitars(guitars)
+
+
+def display_guitars(guitars):
+    """Display formatted list of guitars"""
+    for i, guitar in enumerate(guitars, 1):
+        vintage_string = ""
+        if guitar.is_vintage():
```
Comment (RemoWeiersmuller) [inline prac_06/guitars.py:34]: is it SRP if your printing AND deciding if the guitar is vintage or not?

File: prac_06/guitars.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        cost = get_valid_number(prompt)
+        guitar = Guitar(name, int(year), cost)
+        guitars.append(guitar)
+        print(f"{guitar} added.")
+        name = input("Name: ")
+
+    display_guitars(guitars)
+
+
+def display_guitars(guitars):
```
Comment (RemoWeiersmuller) [inline prac_06/guitars.py:30]: The sample output started with something like this :"These are my guitars:"
you may want to add it. 👍

**Review by RemoWeiersmuller — COMMENTED** [review]
Nice job! Just a few minor things to review. Keep it up! 👍 remember to add a full stop at the end of docs strings.

**Comment by 1428-wsx-f1** [comment]
@RemoWeiersmuller Thank you for the code review 👍

**Pull request merged**